src/rules/no-depends-on-boolean-flags.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+import { RuleCreator } from '@typescript-eslint/utils/eslint-utils';
+import { ancestorsContainsSfCommand, isInCommandDirectory } from '../shared/commands';
+import { isFlag } from '../shared/flags';
+
+export const noDependsOnBooleanFlags = RuleCreator.withoutDocs({
+  meta: {
+    docs: {
+      description: 'Do not allow flags to depend on boolean flags',
+      recommended: 'recommended',
+    },
+    messages: {
+      message: 'Cannot create a flag that `dependsOn` a boolean flag',
```

Comment:
can the message say more, something like, "check use the `relationships` property with a `when` function.  Or maybe the message could print the docs url?